app/services/weather_service.py

```
@@ -28,216 +35,286 @@ def _get_cache(cls) -> ScopedCache:
+            api_key = api_key_30
+        elif api_key_25:
+            # Fallback to current if not historic OR if historic but no 3.0 key
+            provider = "openweather-current"
+            api_key = api_key_25
+            effective_dt = None # For current weather, we don't cache by timestamp
+        elif api_key_30:
+             provider = "openweather-timemachine"
+             api_key = api_key_30
+             effective_dt = target_dt or utc_now() # Timemachine needs a time
+        else:
+            raise ValueError("Configuration error: Unreachable state.")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)